vsintegration/src/FSharp.Editor/CodeFix/AddOpenCodeFixProvider.fs

```diff
@@ -148,7 +148,7 @@ type internal FSharpAddOpenCodeFixProvider
         asyncMaybe {
             let! options = projectInfoManager.TryGetOptionsForEditingDocumentOrProject context.Document
             let! sourceText = context.Document.GetTextAsync(context.CancellationToken)
-            let! parsedInput, checkResults = checker.ParseAndCheckDocument(context.Document, options, sourceText)
+            let! _, parsedInput, checkResults = checker.ParseAndCheckDocument(context.Document, options, sourceText)
```

Comment:
@vasily-kirichenko Or maybe just check the cache again before the real work starts (or as the work progresses), and abandon it if a result has appeared.  The reactor thread doesn't interleave actual work so I suspect you're basically ok?